### PR TITLE
[PPML] Scripts for Graphene on k8s

### PR DESCRIPTION
**Pull request by ColossusChang**
- Added the scripts and yaml files for Graphene on k8s
- Modified `generate-keys.sh` and `generate-password.sh` to generate k8s Secret yaml files

### CODE REVIEW

File: ppml/trusted-realtime-ml/scala/docker-graphene/for-k8s/taskmanager-job-deployment.yaml [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+      component: taskmanager
+  template:
+    metadata:
+      labels:
+        app: flink
+        component: taskmanager
+    spec:
+      containers:
+      - name: taskmanager
+        image: apache/flink:1.13.0-scala_2.11
```
Comment (glorysdj) [inline ppml/trusted-realtime-ml/scala/docker-graphene/for-k8s/taskmanager-job-deployment.yaml:19]: please change to analytics zoo's docker image, and make sure flink JM and TM are deployed in SGX

File: ppml/trusted-realtime-ml/scala/docker-graphene/for-k8s/README.md [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+```bash
+# Configuration and service definition
+kubectl create -f flink-configuration-configmap.yaml
+kubectl create -f jobmanager-service.yaml
+# Create the deployments for the cluster
+kubectl create -f jobmanager-session-deployment.yaml
+kubectl create -f taskmanager-session-deployment.yaml
+```
+
+Both the job manager and the task manager will start automatically when the deployments are created.
```
Comment (glorysdj) [inline ppml/trusted-realtime-ml/scala/docker-graphene/for-k8s/README.md:38]: start automatically in SGX on top of Graphene libos?

File: ppml/trusted-realtime-ml/scala/docker-graphene/for-k8s/README.md [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+kubectl apply -f keys.yaml
+kubectl apply -f password.yaml
+```
+In `master-deployment.yaml`, look for `path_to_start-all-but-flink.sh` and `path_to_enclave-key.pem`, and configure these two paths accordingly. 
+
+Finally, run 
+```bash
+kubectl apply -f master-deployment.yaml
+```
+The components should start on their own, if jobmanager and taskmanager are running.
```
Comment (glorysdj) [inline ppml/trusted-realtime-ml/scala/docker-graphene/for-k8s/README.md:53]: The components including ...

File: ppml/trusted-realtime-ml/scala/docker-graphene/for-k8s/jobmanager-session-deployment.yaml [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+        - name: device-plugin
+          mountPath: /var/lib/kubelet/device-plugins
+        - name: dev-gsgx
+          mountPath: /dev/gsgx
+        - name: aesm-socket
+          mountPath: /var/run/aesmd/aesm.socket
+      volumes:
+      - name: enclave-keys
+        hostPath:
+          path: /home/sdp/yiding/enclave-key/enclave-key.pem
```
Comment (glorysdj) [inline ppml/trusted-realtime-ml/scala/docker-graphene/for-k8s/jobmanager-session-deployment.yaml:73]: replace a mock path and add instruction in readme to tell user to config this

File: ppml/trusted-realtime-ml/scala/docker-graphene/for-k8s/master-deployment.yaml [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+  name: master-deployment
+spec:
+  selector:
+    matchLabels:
+      app: master-app
+  replicas: 1
+  template:
+    metadata:
+      labels:
+        app: master-app
```
Comment (glorysdj) [inline ppml/trusted-realtime-ml/scala/docker-graphene/for-k8s/master-deployment.yaml:13]: can we unify all the app name to analytics-zoo-ppml-trusted-realtime-ml-graphene

**Comment by glorysdj** [comment]
we may also need to add a svc for master to expose 6379 10020 and 10023, etc

**Review by glorysdj — APPROVED** [review]
LGTM

**Comment by qiuxin2012** [comment]
I found a bug here, if `KEYS_PATH` is a relative path, base64 command couldn't find any file.
```
base64: ./keys/keystore.jks: No such file or directory
base64: ./keys/keystore.pkcs12: No such file or directory
base64: ./keys/server.pem: No such file or directory
base64: ./keys/server.crt: No such file or directory
base64: ./keys/server.csr: No such file or directory
base64: ./keys/server.key: No such file or directory
```

**Pull request merged**